server/src/main/java/org/opensearch/search/aggregations/metrics/SumAggregatorFactory.java

```diff
@@ -69,11 +66,23 @@ static void registerAggregators(ValuesSourceRegistry.Builder builder) {
         builder.register(
             SumAggregationBuilder.REGISTRY_KEY,
             List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.DATE, CoreValuesSourceType.BOOLEAN),
-            SumAggregator::new,
+            SumAggregatorFactory::methodName,
             true
         );
     }
 
+    private static Aggregator methodName(String s, ValuesSourceConfig valuesSourceConfig, SearchContext searchContext, Aggregator aggregator, Map<String, Object> stringObjectMap) throws IOException {
+        if (valuesSourceConfig.hasValues()) {
```

Comment:
nit: I really prefer to flatten if-chains by returning early.   Something like: 

```
if (!valuesSourceConfig.hasValues()) { 
  return new SumNullAggregator(s, valuesSourceConfig, searchContext, aggregator, stringObjectMap);
}

// use kahan summation aggregator for floating point numbers
if (((ValuesSource.Numeric) valuesSourceConfig.getValuesSource()).isFloatingPoint()) {
  return new SumAggregator(s, valuesSourceConfig, searchContext, aggregator, stringObjectMap);
}

return new SumIntegralAggregator(s, valuesSourceConfig, searchContext, aggregator, stringObjectMap);

```

Reply:
nit has been picked in [ced4530](https://github.com/matthewryanwells/OpenSearch/pull/1/commits/ced4530d9a0505db767d5351805355657b99653e)